src/mixins/ERC4626.sol

```diff
@@ -59,48 +59,44 @@ abstract contract ERC4626 is ERC20 {
         afterDeposit(underlyingAmount);
     }
 
-    function withdraw(address to, uint256 underlyingAmount) public virtual returns (uint256 shares) {
-        _burn(msg.sender, shares = calculateShares(underlyingAmount));
+    function mint(address to, uint256 shareAmount) public virtual returns (uint256 underlyingAmount) {
+        _mint(to, shareAmount);
 
-        emit Withdraw(msg.sender, to, underlyingAmount);
+        emit Deposit(msg.sender, to, underlyingAmount = calculateUnderlying(shareAmount));
```

Comment:
This is more of a code style preference but I would prefer it if `underlyingAmount = calculateUnderlying(shareAmount)` was defined outside of the `emit`, it is more explicit in my opinion seeing as we are using the underlyingAmount in multiple places after the event as well.

Reply:
(see comment below, this is actually part of a bug I think)